pyanaconda/modules/payloads/payload/dnf/kickstart_converter.py

```diff
@@ -0,0 +1,83 @@
+#
+# Converter from DNF ks data to payload sources.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pykickstart.errors import KickstartParseError
+
+from pyanaconda.core.i18n import _
+from pyanaconda.modules.payloads.constants import SourceType
+from pyanaconda.modules.payloads.source.factory import SourceFactory
+
+
+class KickstartConverter():
+    """Process kickstart including checks and return source type."""
+
+    @classmethod
+    def convert(cls, ks_data):
+        """Create source based on the KS data passed.
+
+        This will do three things:
+        1) Process KS data and check for validity.
+        2) Create source object based on the data.
+        3) Pass kickstart data into the created source.
+
+        :param ks_data: kickstart data for DNF module
+        :return: instance of source object
+        :raise: ParseKickstartError if kickstart data are not correct
+        """
+        source_type = cls._process_kickstart(ks_data)
```

Comment:
This implementation is pretty much the same as `get_type_for_kickstart` from `PayloadFactory`. Because of that, I suggest to move the method `_process_kickstart` to `SourceFactory`, rename it to `get_rpm_type_for_kickstart` and move the content of the method `convert` to the method `process_kickstart` of the class `DNFModule`. It shouldn't cause any new issues and we will move closer to the "common practice".

Reply:
I don't think it is a good way to go but yeah we've already went to this rabbit hole and it's too late to change it here.
We should definitely investigate the possibility to move kickstart processing to a separate space for the module. It's a shame that most of the code in Payload have to deal with KS in some way.

But I agree that we should stay consistent with an existing solution and change this later.